### DISCUSSION TITLE
Conditional validation rules.

### DISCUSSION THREAD

**Discussion by nortmas** [spatie/laravel-data#603] [Ideas]
It would be great if it was possible to specify the rule or create a custom rule that may depend on the values from the other parameters. For example:

```
#[Required]
public int $day; 

#[Required]
public string $condition; 

#[ConditionalRule(MyCustomConditionalRule::class, ['day', 'condition'])]
public string $price; 
```
where: 

```
MyCustomConditionalRule implements ConditionalRule {
     public function validate(mixed $values, Closure $fail): void
    {
       $valid = true;
       if ($values['day'] > 5 && $values['condition'] === 'some_value') {
           $valid = $price > 1000;
       }

       // some other logic...

        if (! $valid) {
            $fail(__('validation.phone_number'));
        }
    }
}

I faced with this use case / limitation many times and in such cases I have to use native laravel validation.

**Comment by rubenvanassche** [comment]
Use the static rules method in such case with the ValidationContext, you'll find the values in there and then could pass them to a custom rule constructor.

**Reply by striker4150** [reply]
I would really appreciate a way to properly implement parameters in a custom validation rule. The way I see it, validation rules are already being applied through attributes and rule inferrers. We should not have to override all of that through the static rules method whenever one bit of validation needs to be customized. It seems to go against the idea of using attributes or rule inferrers in the first place if you're going to have to rewrite all the rules in the static rules method anyways. The static rules method should be the last resort - only used when you need absolute control over all the rules being set, not when you need to customize a single rule. This is just my opinion though.

**Reply by rubenvanassche** [reply]
You're always welcome to PR such a thing
